examples/generative/pixelcnn.py

```diff
@@ -0,0 +1,204 @@
+"""
+Title: PixelCNN
+Author: ADMoreau
+Date created: 2020/05/17
+Last modified: 2020/05/20
+Description: PixelCNN implemented in Keras.
+"""
+
+"""
+## Setup
+"""
+
+import numpy as np
+import tensorflow as tf
+from tensorflow import keras, nn
+from tensorflow.keras import layers
+from tensorflow.python.ops import nn_ops
+
+"""
+#Getting the Data
+"""
+
+# Model / data parameters
+num_classes = 10
+input_shape = (28, 28, 1)
+n_residual_blocks = 5
+# the data, split between train and test sets
+(x, _), (y, _) = keras.datasets.mnist.load_data()
+# Concatenate all of the images together
+data = np.concatenate((x, y), axis=0)
+# round all pixel values less than 33% of the max 256 value to 0
+# anything above this value gets rounded up to 1 so that all values are either
+# 0 or 1
+data = np.where(data < (0.33 * 256), 0, 1)
+
+"""
+#Create two classes for the requisite Layers for the model
+"""
+
+# the first layer to create will be the PixelCNN layer, this layer is simply
+# the 2D convolutional layer with the masking included
+class PixelConvLayer(keras.layers.Conv2D):
+    def __init__(self, mask_type, *args, **kwargs):
+        super(PixelConvLayer, self).__init__(*args, **kwargs)
+        self.mask_type = mask_type
+        self.mask = None
+
+    def call(self, inputs):
+        if self._recreate_conv_op(inputs):
```

Comment:
That way, you won't have to touch `_convolution_op`, which is private

(note that alternatively you could have called `super(self, PixelConvLayer).call()`)